### PR TITLE
Bm 109

### PR DESCRIPTION
**Pull request by greg-fti**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -121,6 +122,21 @@ _note: if .dropOffDate is not set, it will be calculated with .pickUpDate + .dur
 |         | .dateFrom    | '20092017' 
 |         | .dateTo      | '20092017' 
 
+
+| type      | fields       | example
+| :---      | :---         | :---
+|'roundTrip'| .bookingId   | 'NEZE2784NQXTHEN' 
+|           | .destination | 'YYZ' 
+|           | .no          | '1'
```
Comment (EyesOnlyNet) [inline README.md]: we should give that property a more readable name - what does this property mean - number of what?

File: README.md [hunk 2]
```diff
@@ -121,6 +122,21 @@ _note: if .dropOffDate is not set, it will be calculated with .pickUpDate + .dur
...
 
+
+| type      | fields       | example
+| :---      | :---         | :---
+|'roundTrip'| .bookingId   | 'NEZE2784NQXTHEN' 
+|           | .destination | 'YYZ' 
+|           | .no          | '1' 
+|           | .startDate   | '05122017' 
+|           | .endDate     | '16122017'
+|           | .title       | 'H'
```
Comment (EyesOnlyNet) [inline README.md]: the following fields are not part of a service. they are parts of the "customer data". we should separate them from the service data.

File: src/crsAdapter/TomaAdapter.js [hunk 3]
```diff
@@ -305,6 +313,33 @@ class TomaAdapter {
...
+     */
+    mapRoundTripServiceFromXmlObjectToAdapterObject(xml, lineNumber) {
+        let service = {
+            type: SERVICE_TYPES.roundTrip,
+            bookingId: xml['ServiceCode.' + lineNumber],
+            destination: xml['Accommodation.' + lineNumber],
+            no: xml['Count.' + lineNumber],
+            startDate: moment(xml['From.' + lineNumber], CONFIG.crs.dateFormat).format(this.options.useDateFormat),
+            endDate: moment(xml['To.' + lineNumber], CONFIG.crs.dateFormat).format(this.options.useDateFormat),
+            title: xml['Title.' + lineNumber],
```
Comment (EyesOnlyNet) [inline src/crsAdapter/TomaAdapter.js]: to be correct `xml['Title.' + lineNumber]` is not really a title - it is the "salutation". title is something like "dr." or "M.Sc."

File: src/crsAdapter/TomaAdapter.js [hunk 4]
```diff
@@ -488,6 +527,25 @@ class TomaAdapter {
...
+    assignRoundTripServiceFromAdapterObjectToXmlObject(service, xml, lineNumber) {
+        xml['KindOfService.' + lineNumber] = CONFIG.crs.serviceTypes.roundTrip;
+        xml['ServiceCode.' + lineNumber] = service.bookingId;
+        xml['Accommodation.' + lineNumber] = service.destination;
+        xml['Count.' + lineNumber] = service.no;
+        xml['From.' + lineNumber] = moment(service.startDate, this.options.useDateFormat).format(CONFIG.crs.dateFormat);
+        xml['To.' + lineNumber] = moment(service.endDate, this.options.useDateFormat).format(CONFIG.crs.dateFormat);
+        xml['Title.' + lineNumber] = service.title;
+        xml['Name.' + lineNumber] = service.name;
+        xml['Reduction.' + lineNumber] = service.birthday ? service.birthday : service.age;
```
Comment (EyesOnlyNet) [inline src/crsAdapter/TomaAdapter.js]: you can use a shorter version: `service.birthday || service.age` it is the same like in PHP the `service.birthday ?: service.age`

File: tests/unit/crsAdapter/TomaAdapter.test.js [hunk 5]
```diff
@@ -224,6 +224,73 @@ describe('TomaAdapter', () => {
...
+
+            expect(adapter.getData()).toEqual({
+                services: [
+                    roundTripService1,
+                ]
+            });
+
+            TomaConnection.GetXmlData.and.returnValue(xml2);
+
+            expect(adapter.getData()).toEqual({
```
Comment (EyesOnlyNet) [inline tests/unit/crsAdapter/TomaAdapter.test.js]: if you test different use cases, then separate the tests and give them a name which infers the purpose of the test

File: README.md (hunk 1 above)
Comment (greg-fti) [reply README.md]: As you suggested, I have written an email to afu@nezasa.com to find out, what this mystery things is.

Comment (EyesOnlyNet) [reply README.md]: :+1:

**Comment by EyesOnlyNet** [comment]
@greg-fti when you merge this pull request, please use "squash and merge"

**Pull request merged**